### PR TITLE
Re-add flag for ocb build to enable version number

### PR DESCRIPTION
**Pull request by jackgopack4**
**Description**
This re-enables the functionality to set a version number for ocb binary (cmd/builder). It was erroneously removed.

**Link to tracking issue**
Along with https://github.com/open-telemetry/opentelemetry-collector/pull/11208, closes https://github.com/open-telemetry/opentelemetry-collector-releases/issues/664.

**Testing**
ran build and release cycles inside personal fork repositories (jackgopack4/opentelemetry-collector and jackgopack4/opentelemetry-collector-releases)

**Documentation**
none, small bugfix

### CODE REVIEW

**Review by mx-psi — APPROVED** [review]

**Comment by mx-psi** [comment]
Do we need https://github.com/open-telemetry/opentelemetry-collector/pull/11208 before merging this?

**Comment by chan-tim-sumo** [comment]
hey @bogdandrutu can you review this PR also? currently blocked from this since this is needed to get a new release out for customers from my side. thanks!

**Review by atoulme — APPROVED** [review]

**Comment by chan-tim-sumo** [comment]
hey @TylerHelmuth can you take a look at this PR? currently blocked from this since this is needed to get a new release out for customers from my side. thanks!

File: cmd/builder/.goreleaser.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ builds:
   - flags:
       - -trimpath
     ldflags:
-      - -s -w
+      - -s -w -X go.opentelemetry.io/collector/cmd/builder/internal.version={{ .Version }}
```
Comment (TylerHelmuth) [inline cmd/builder/.goreleaser.yml:11]: @jackgopack4 have you been able to test this change?

Comment (TylerHelmuth) [reply cmd/builder/.goreleaser.yml:11]: nvm, I forgot this is the same string it used to be

**Review by TylerHelmuth — APPROVED** [review]

File: cmd/builder/.goreleaser.yml (hunk 1 above)
Comment (jackgopack4) [reply cmd/builder/.goreleaser.yml:11]: yes, though, I did a few releases in github.com/jackgopack4/opentelemetry-collector-releases
https://github.com/jackgopack4/opentelemetry-collector-releases/releases/tag/cmd%2Fbuilder%2Fv0.107.1
for example

**Pull request merged**